src/Task_21/PersonService.java

```diff
@@ -0,0 +1,94 @@
+package Task_21;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class PersonService {
+
+    public static Set<Person> people = new TreeSet<>(new FirstComparator());
+    public static String strInfo;
+    public static int quantityOfWomen = 0;
+    public static int quantityOfMen = 0;
+    public static int quantityWithAgeMoreThan30 = 0;
+
+    public static void readFromFile(String filePath) {
+
+        try {
```

Comment:
The outer `try` block shouldn't be here.
It should be only one try-catch block

Reply:
If we create only one try, the program doesn't give us an opportunity to put close() methods into finally block.
![image](https://user-images.githubusercontent.com/70838761/219049959-137f0ba0-7374-4635-b4ab-36e4bf911056.png)

Reply:
Is it correct in this case?
![image](https://user-images.githubusercontent.com/70838761/219077962-17521784-eb7b-4f2b-9a08-cc14f870b79c.png)
![image](https://user-images.githubusercontent.com/70838761/219078173-93efab06-c98c-4934-a1d1-875853ab8155.png)

Reply:
Absolutely